src/app/services/reducers/permission-mode-reducer.ts

```diff
@@ -15,3 +17,14 @@ export function permissionModeType(state: PERMISSION_MODE_TYPE = PERMISSION_MODE
             return state;
     }
 }
+
+export function hideDialog(state: boolean = false, action: IAction): any {
+    switch (action.type) {
+        case CLOSE_POP_UP_SUCCESS:
+            return action.response
+        case PREPARE_NEXT_POP_UP_SUCCESS:
+            return action.response
+        default:
```

Comment:
Since you're returning action.response no matter the action type, maybe you don't need the two actions? The two action types could be reduced to just `DISPLAY_POPUP`. Kind of like the `permissionsPanelOpen` reducer. (But if it's easier for you to have two action types you can keep it as two).

Reply:
97d5789597b77a9f4963ddd78326fb10af34b004